src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/NetworkInterfaceTests.ps1

```diff
@@ -887,3 +887,98 @@ function Test-NetworkInterfaceWithAcceleratedNetworking
+        $job | Wait-Job
+        $expectedNic = Get-AzureRmNetworkInterface -Name $nicName -ResourceGroupName $rgname
+
+        # Create required dependencies (Vtap)
+        $DestinationEndpoint = $expectedNic.IpConfigurations[0]
+        $actualVtap = New-AzureRmVirtualNetworkTap -ResourceGroupName $rgname -Name $vtapName -Location $location -DestinationNetworkInterfaceIPConfiguration $DestinationEndpoint  -Force
+        $vVirtualNetworkTap = Get-AzureRmVirtualNetworkTap -ResourceGroupName $rgname -Name $vtapName;
+
+        # Create source Nic which is getting tapped
+        $sourceIpConfig = New-AzureRmNetworkInterfaceIpConfig -Name $sourceIpConfigName -Subnet $vnet.Subnets[0]
+        $sourceNic = New-AzureRmNetworkInterface -Name $sourceNicName -ResourceGroupName $rgname -Location $location -IpConfiguration $sourceIpConfig -Tag @{ testtag = "testval" }
+
+        # Add tap configuration
+        Add-AzureRmNetworkInterfaceTapConfig -NetworkInterface $sourceNic -VirtualNetworkTap $vVirtualNetworkTap -Name $rname
+
+        #get tap configuration
+        $tapConfig = Get-AzureRmNetworkInterfaceTapConfig -ResourceGroupName $rgname -NetworkInterfaceName $sourceNicName -Name $rname
+        Assert-NotNull $tapConfig
+        Assert-AreEqual $tapConfig.ResourceGroupName $rgname
+        Assert-AreEqual $tapConfig.NetworkInterfaceName $sourceNicName
+        Assert-AreEqual $tapConfig.Name $rname
+
+        # get nic and check back reference        
+        $sourceNic = Get-AzureRmNetworkInterface -Name $sourceNicName -ResourceGroupName $rgname
+        Assert-NotNull $sourceNic.TapConfigurations
+        Assert-NotNull $sourceNic.TapConfigurations[0]
+        Assert-AreEqual $sourceNic.TapConfigurations[0].Id $tapConfig.Id
+
+        # get vtap and check back reference
+        $vVirtualNetworkTap = Get-AzureRmVirtualNetworkTap -ResourceGroupName $rgname -Name $vtapName;
+        Assert-NotNull $vVirtualNetworkTap.NetworkInterfaceTapConfigurations
+        Assert-NotNull $vVirtualNetworkTap.NetworkInterfaceTapConfigurations[0]
+        Assert-AreEqual $vVirtualNetworkTap.NetworkInterfaceTapConfigurations[0].Id $tapConfig.Id
+
+   
+        # Remove NetworkInterfaceTapConfiguration
+        $removeNetworkInterfaceTapConfiguration = Remove-AzureRmNetworkInterfaceTapConfig -ResourceGroupName $rgname -NetworkInterfaceName $sourceNicName -Name $rname -PassThru -Force;
+        Assert-AreEqual $true $removeNetworkInterfaceTapConfiguration;
+
+        $sourceNic = Get-AzureRmNetworkInterface -Name $sourceNicName -ResourceGroupName $rgname
+        Assert-NotNull $sourceNic.TapConfigurations
+        Assert-Null $sourceNic.TapConfigurations[0]
+
+        # get vtap and check back reference
+        $vVirtualNetworkTap = Get-AzureRmVirtualNetworkTap -ResourceGroupName $rgname -Name $vtapName;
```

Comment:
From destination nic, we should also check all the taps and see if they are matching.

Reply:
Not sure if I understood you....I’m validating source Nic tap configurations and Vtap back reference...